src/VisualStudio/Core/Test.Next/Services/LspDiagnosticsTests.cs

```diff
@@ -38,6 +40,7 @@ public class LspDiagnosticsTests : AbstractLanguageServerProtocolTests
         public async Task AddDiagnosticTestAsync()
         {
             using var workspace = CreateTestWorkspace("", out _);
+            var listenerProvider = workspace.ExportProvider.GetExportedValue<IAsynchronousOperationListenerProvider>();
```

Comment:
is it possible to update RunPublishDiagnosticsAsync to just retrieve the listenerProvider from the export provider itself, instead of doing it in every test?

Reply:
Done!